src/Components/Address/useAddressAutocomplete.ts

```diff
@@ -54,37 +72,35 @@ export const useAddressAutocomplete = (
     }
   }, [isUSAddress, result.length])
 
-  const fetchSuggestions = useCallback(
-    async ({ search, selected }: { search: string; selected?: string }) => {
-      const params = {
-        key: apiKey,
-        search: search,
+  const fetchSuggestions = useMemo(() => {
+    const throttledFetch = throttle(
```

Comment:
switched from `useCallback` to `useMemo()` and dropped the intermediate function declaration in favor of building it inside the body to `useMemo`. Chose `useMemo` because of an error from the linter saying that i should pass an inline function, and not recognizing that `throttle(() => function i had moved in from the external fetchSuggestions...` was pretty much inline. In practice i think this should work the same, but that's the reason.

Reply:
With useCallback however, tests did start failing (even when i used an empty dependency array). The point of this test was to assert `throttle()` is only called once so we don't start creating duplicate throttles, which would break the functionality:

![image](https://github.com/artsy/force/assets/9088720/516f0b44-9484-4134-8845-0ae632a8acb0)

Reply:
Yep using memo for both debounce and throttle is preferrable.